pkgs/os-specific/darwin/apple-source-releases/adv_cmds/package.nix

```diff
@@ -76,25 +112,53 @@ mkAppleDerivation {
 
   nativeBuildInputs = [
     bison
+    bmake
     flex
     pkg-config
   ];
 
+  enableParallelBuilding = true;
+
   mesonFlags = [
     # Even though adv_cmds is built with a newer SDK, the default SDK is still the deployment target.
     # Don’t build packages that use newer APIs unnecessarily.
     (lib.mesonOption "sdk_version" (lib.getVersion apple-sdk))
   ];
 
-  postBuild = ''
-    # Build the locales TODO
-  '';
+  buildPhase =
+    ''
+      ninjaBuildPhase
+    ''
+    +
+      # Cross-compilation requires using adv_cmds from the buildPlatform.
+      (
+        if stdenvNoCC.buildPlatform.canExecute stdenvNoCC.hostPlatform then
+          ''
+            export PATH=$PWD:$PATH
+          ''
+        else
+          ''
+            export PATH=$PWD:${lib.getBin buildPackages.darwin.adv_cmds}/bin
+          ''
+      )
```

Comment:
Actually, why don’t we just split this locale stuff back into its own package? This clearly just wants a `nativeBuildInputs = [ darwin.adv_cmds ];`, we’re having to compose two separate build systems in one derivation, and one of them goes to a different output anyway. I think it would be a lot nicer to just have a separate package for the locale data.

Reply:
It’s not unusual for a package that needs to generate some data to have special handling for cross-compilation (e.g., ICU and ncurses). I can move it to a passthru derivation, so `adv_cmds.locale` works as it did before.